### PR TITLE
Offboarding Clusters from Cluster-registry

### PR DESCRIPTION
**Pull request by vivekzhere**
- Block deletion of SFCluster and secret
- Allow deletion if no instance provisioned
- Add tests
- Fix `go lint` errors

Fetaure: HCPCFS-2957

### CODE REVIEW

**Review by anoopjb — APPROVED** [review]
LGTM.
There are some staticcheck failures.

**Comment by vinaybheri** [comment]
Validation on k8s cluster: n is succeeded.

**Comment by vinaybheri** [comment]
Validation on k8s cluster: n-1 is succeeded.

**Comment by vinaybheri** [comment]
Validation on k8s cluster: n is succeeded.

**Comment by vinaybheri** [comment]
Validation on k8s cluster: n-2 is succeeded.

File: interoperator/controllers/multiclusterdeploy/offboarding/offboarding_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+)
+
+/*
+The first step to writing a simple integration test is to actually create an instance of CronJob you can run tests against.
+Note that to create a CronJob, you’ll need to create a stub CronJob struct that contains your CronJob’s specifications.
+
+Note that when we create a stub CronJob, the CronJob also needs stubs of its required downstream objects.
+Without the stubbed Job template spec and the Pod template spec below, the Kubernetes API will not be able to
+create the CronJob.
+*/
```
Comment (jayeshjk) [inline interoperator/controllers/multiclusterdeploy/offboarding/offboarding_controller_test.go:39]: unrelated comments.

**Review by jayeshjk — APPROVED** [review]
A minor comment. LGTM

File: interoperator/controllers/multiclusterdeploy/offboarding/offboarding_controller_test.go (hunk 1 above)
Comment (vivekzhere) [reply interoperator/controllers/multiclusterdeploy/offboarding/offboarding_controller_test.go]: Removed the comments

**Pull request merged**